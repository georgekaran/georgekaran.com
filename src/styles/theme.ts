export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem'
  },
  border: {
    radius: '0.4rem'
  },
  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    medium: 500,
    bold: 600,
    sizes: {
      xsmall: '0.8rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '2.4rem',
      xlarge: '3.2rem',
      xxlarge: '4.2rem',
      xxxlarge: '5rem'
    }
  },
  colors: {
    green: '#28F5BE',
    black: '#1F1F24',
    green01: '#1BE0AB',
    green02: '#43FECB',
    black01: '#27272D',
    black02: '#35353D',
    black03: '#484854',
    black04: '#737380',
    purple: '#6321CE',
    pink: '#FF3773',
    blue: '#3B37FF',
    gradient01: 'linear-gradient(270.03deg, #3B37FF 0.05%, rgba(59, 55, 255, 0) 98.15%)',
    gradient02: 'linear-gradient(265.22deg, #28F5BE -1.83%, rgba(40, 245, 190, 0) 95.01%)',
    gradient03: 'linear-gradient(83.89deg, #28F5BE 2.81%, #3B37FF 97.94%)',
    white: '#FFFFFF',
    gray: '#D4D4D4'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
}
