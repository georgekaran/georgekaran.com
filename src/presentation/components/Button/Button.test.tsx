import React from 'react'
import { screen, RenderResult } from '@testing-library/react'
import { AddShoppingCart } from '@styled-icons/material-outlined/AddShoppingCart'

import { Button, ButtonProps, ButtonTypes } from '.'
import { renderWithTheme } from '@/test/helpers'
import theme from '@/presentation/styles/theme'

type SutProps = Omit<ButtonProps, 'children'> & ButtonTypes

const makeSut = ({ size = 'medium', fullWidth = false, icon = null, as = 'button', ...props }: SutProps = {}): RenderResult => {
  return renderWithTheme(<Button size={size} fullWidth={fullWidth} icon={icon} as={as} {...props}>Button</Button>)
}

describe('<Button />', () => {
  test('should render with correct initial state', () => {
    const { container } = makeSut()
    expect(container.firstChild).toMatchSnapshot()
    const button = screen.getByRole('button', { name: /Button/ })
    expect(button).toHaveStyleRule('height', '4rem')
    expect(button).toHaveStyleRule('font-size', theme.font.sizes.small)
    expect(button).toHaveStyleRule('padding', `${theme.spacings.xxsmall} ${theme.spacings.medium}`)
  })

  test('should render small if small size is provided', () => {
    makeSut({ size: 'small' })
    const button = screen.getByRole('button', { name: /Button/ })
    expect(button).toHaveStyleRule('height', '3rem')
    expect(button).toHaveStyleRule('font-size', theme.font.sizes.small)
    expect(button).toHaveStyleRule('padding', theme.spacings.xxsmall)
  })

  test('should render large if large size is provided', () => {
    makeSut({ size: 'large' })
    const button = screen.getByRole('button', { name: /Button/ })
    expect(button).toHaveStyleRule('height', '5rem')
    expect(button).toHaveStyleRule('font-size', theme.font.sizes.medium)
    expect(button).toHaveStyleRule('padding', `${theme.spacings.xxsmall} ${theme.spacings.xlarge}`)
  })

  test('should render fullWidth if prop is provided', () => {
    makeSut({ fullWidth: true })
    const button = screen.getByRole('button', { name: /Button/ })
    expect(button).toHaveStyleRule('width', '100%')
  })

  test('should render icon if an icon is provided', () => {
    makeSut({ icon: <AddShoppingCart data-testid="icon" /> })
    const icon = screen.getByTestId('icon')
    expect(icon).toBeInTheDocument()
    expect(screen.getByText(/Button/)).toBeInTheDocument()
  })

  test('should render as anchor if prop \'as\' is a', () => {
    makeSut({ as: 'a', href: '/link' })
    const buttonAsLink = screen.getByRole('link', { name: /Button/i })
    expect(buttonAsLink).toBeInTheDocument()
  })
})
