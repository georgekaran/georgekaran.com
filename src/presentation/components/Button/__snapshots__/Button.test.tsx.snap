// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Button /> should render with correct initial state 1'] = `
.c0 {
  background: #28F5BE;
  color: #27272D;
  font-weight: 500;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.6rem;
  padding: 0.8rem 3.2rem;
}

.c0:hover {
  background: #1BE0AB;
}

<button
  class="c0"
>
  <span>
    Button
  </span>
</button>
`
